src/board/mod.rs

```diff
@@ -243,19 +257,90 @@ impl Board {
             let color = self.next_player();
             self.vacant
                 .iter()
-                .map(|coord| Play(color.clone(), coord.col, coord.row))
+                .map(|coord| Play(color, coord.col, coord.row))
                 .filter(|m| self.is_legal(*m).is_ok())
                 .collect()
         }
     }
-
+    
+    pub fn playout_move(&self, rng: &mut XorShiftRng) -> Move {
+            let color = self.next_player();
+            let vacant = self.vacant();
+            let mut iter = vacant
+                .iter()
+                .map(|coord| Play(color, coord.col, coord.row))
+                .filter(|m| !self.is_eye(&m.coord(), *m.color()) && self.playout_legal_move(*m) )
+                .peekable();
+            //let count = iter.count();
+            
+            if iter.is_empty() {
+                let color = self.next_player();
+                Pass(color)
+            } else {
+                //while loop testing all vacant spots
+
+                loop {
+                    let random_vacant = vacant[rng.gen::<usize>() % vacant.len()];
+                    let random_play = Play(color, random_vacant.col, random_vacant.row);
+                    if !self.is_eye(&random_vacant, color) && self.playout_legal_move(random_play) {
+                    	return random_play;
+                	} 
+                }
+            }
+	}
+    
+    pub fn playout_moves(&self) -> Vec<Move> {
+        	let color = self.next_player();
+            self.vacant
+                .iter()
+                .map(|coord| Play(color, coord.col, coord.row))
+        	    .filter(|m| !self.is_eye(&m.coord(), *m.color()) && self.playout_legal_move(*m) )
+                .collect()
+	}
```

Comment:
Doesn't this duplicate the code from `legal_moves`? If so, why are we doing this?